src/wp-admin/includes/plugin.php

```diff
@@ -1203,26 +1203,55 @@ function validate_plugin_requirements( $plugin ) {
 	}
 
 	if ( WP_Plugin_Dependencies::has_unmet_dependencies( $plugin ) ) {
-		$dependencies       = WP_Plugin_Dependencies::get_dependencies( $plugin );
-		$unmet_dependencies = array();
+		$dependency_names       = WP_Plugin_Dependencies::get_dependency_names( $plugin );
+		$unmet_dependencies     = array();
+		$unmet_dependency_names = array();
 
-		foreach ( $dependencies as $dependency ) {
+		foreach ( $dependency_names as $dependency => $dependency_name ) {
 			$dependency_file = WP_Plugin_Dependencies::get_dependency_filepath( $dependency );
 
 			if ( false === $dependency_file ) {
-				$unmet_dependencies['not_installed'][] = $dependency;
+				$unmet_dependencies['not_installed'][ $dependency ] = $dependency_name;
+				$unmet_dependency_names[]                           = $dependency_name;
 			} elseif ( is_plugin_inactive( $dependency_file ) ) {
-				$unmet_dependencies['inactive'][] = $dependency;
+				$unmet_dependencies['inactive'][ $dependency ] = $dependency_name;
+				$unmet_dependency_names[]                      = $dependency_name;
 			}
 		}
 
+		$error_message = sprintf(
+			/* translators: 1: Plugin name, 2: Number of plugins, 3: A comma-separated list of plugin names. */
+			_n(
+				'<strong>Error:</strong> %1$s requires %2$d plugin to be installed and activated: %3$s.',
+				'<strong>Error:</strong> %1$s requires %2$d plugins to be installed and activated: %3$s.',
+				count( $unmet_dependency_names )
+			),
+			$plugin_headers['Name'],
+			count( $unmet_dependency_names ),
+			implode( wp_get_list_item_separator(), $unmet_dependency_names )
+		);
+
+		if ( is_multisite() ) {
+			if ( current_user_can( 'manage_network_plugins' ) ) {
+				$error_message .= ' ' . sprintf(
+					/* translators: %s: Link to the plugins page. */
+					__( '<a href="%s">Manage plugins</a>' ),
```

Comment:
Nit: either add a period at the end for consistency, or move the markup out of the string entirely. The former should be ok though


```suggestion
					__( '<a href="%s">Manage plugins</a>.' ),
```

Reply:
In WP-CLI, the message will end in `. .` because WP-CLI strips the link. It's not a huge issue, but was the reason I omitted the period. What do you think?